source/ingress_fqdn_test.go

```diff
@@ -0,0 +1,362 @@
+		},
+	} {
+		t.Run(tt.title, func(t *testing.T) {
+			_, err := NewIngressSource(
+				t.Context(),
+				fake.NewClientset(),
+				"",
+				"",
+				tt.fqdnTemplate,
+				false,
+				false,
+				false,
+				false,
+				labels.Everything(),
+				[]string{},
+			)
+
+			if tt.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestIngressSourceFqdnTemplatingExamples(t *testing.T) {
+
+	for _, tt := range []struct {
+		title        string
+		ingresses    []*networkv1.Ingress
+		fqdnTemplate string
+		expected     []*endpoint.Endpoint
+	}{
+		{
+			title: "templating resolve Ingress source hostnames to IP",
+			ingresses: []*networkv1.Ingress{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "my-ingress",
+						Namespace: "default",
+					},
+					Spec: networkv1.IngressSpec{
+						IngressClassName: testutils.StringPtr("my-ingress"),
+						Rules: []networkv1.IngressRule{
+							{
+								Host: "example.org",
+								IngressRuleValue: networkv1.IngressRuleValue{
+									HTTP: &networkv1.HTTPIngressRuleValue{
+										Paths: []networkv1.HTTPIngressPath{
+											{
+												Backend: networkv1.IngressBackend{
+													Service: &networkv1.IngressServiceBackend{
+														Name: "my-service",
+														Port: networkv1.ServiceBackendPort{
+															Name: "http",
+														},
+													},
+												},
+												PathType: testutils.ToPtr(networkv1.PathTypePrefix),
+												Path:     "/",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					Status: networkv1.IngressStatus{
+						LoadBalancer: networkv1.IngressLoadBalancerStatus{
+							Ingress: []networkv1.IngressLoadBalancerIngress{
+								{Hostname: "10.200.130.84.nip.io"},
+							},
+						},
+					},
+				},
+			},
+			fqdnTemplate: "{{.Name }}.nip.io",
+			expected: []*endpoint.Endpoint{
+				{DNSName: "example.org", RecordType: endpoint.RecordTypeCNAME, Targets: endpoint.Targets{"10.200.130.84.nip.io"}},
+				{DNSName: "my-ingress.nip.io", RecordType: endpoint.RecordTypeCNAME, Targets: endpoint.Targets{"10.200.130.84.nip.io"}},
+			},
+		},
+		{
+			title: "templating resolve hostnames with nip.io",
+			ingresses: []*networkv1.Ingress{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "my-ingress",
+						Namespace: "default",
+					},
+					Spec: networkv1.IngressSpec{
+						IngressClassName: testutils.StringPtr("my-ingress"),
+						Rules: []networkv1.IngressRule{
+							{Host: "example.org"},
+						},
+					},
+					Status: networkv1.IngressStatus{
+						LoadBalancer: networkv1.IngressLoadBalancerStatus{
+							Ingress: []networkv1.IngressLoadBalancerIngress{
+								{Hostname: "10.200.130.84.nip.io"},
+							},
+						},
+					},
+				},
+			},
+			fqdnTemplate: "{{ range .Status.LoadBalancer.Ingress }}{{ if contains .Hostname \"nip.io\" }}example.org{{end}}{{end}}",
```

Comment:
```suggestion
			fqdnTemplate: `{{ range .Status.LoadBalancer.Ingress }}{{ if contains .Hostname "nip.io" }}example.org{{end}}{{end}}`,
```
To avoid `"` escape?